### PR TITLE
AMP-71142 Hotjar plugin docs

### PR DESCRIPTION
**Pull request by jooohhn**
# Amplitude Developer Docs PR

## Description

- Adds new section to Plugin docs regarding how to forward events to Hotjar

## Deadline

Important, not urgent

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [x] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/ampli/plugin.md [hunk 1]
```diff
@@ -2,38 +2,209 @@
 title: Plugin
 description: Use plugins to extend Amplitude by running a sequence of custom code on every event. This pattern is flexible and you can use it to support event enrichment, transformation, filtering, routing to third-party destinations, and more.
 template: guide.html
-
 ---
 
 !!!note
-    Plugins are supported in the latest version of Ampli when used in conjunction with the [new Amplitude browser SDKs](https://www.docs.developers.amplitude.com/data/sdks/typescript-browser/migration/). If you are using an older version of Ampli or a [Maintenance Amplitude SDK](https://www.docs.developers.amplitude.com/data/sdks/typescript-browser/migration/), see **[Middleware](../middleware/)**.
+Plugins are supported in the latest version of Ampli when used in conjunction with the [new Amplitude browser SDKs](https://www.docs.developers.amplitude.com/data/sdks/typescript-browser/migration/). If you are using an older version of Ampli or a [Maintenance Amplitude SDK](https://www.docs.developers.amplitude.com/data/sdks/typescript-browser/migration/), see **[Middleware](../middleware/)**.
```
Comment (Mercy811) [inline docs/data/ampli/plugin.md:8]: Please revert this change. This is the syntax for [admonitioins](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage). And also it helps to do a [self review ](https://github.com/Amplitude-Developer-Docs/amplitude-dev-center/wiki#3-run-a-local-server-and-preview-your-changes)of all chnages.

File: docs/data/ampli/plugin.md [hunk 2]
```diff
@@ -2,38 +2,209 @@
...
+## Destination type plugin
+
+Here's an example of a plugin that sends each event that's instrumented to a target server URL using your preferred HTTP client.
+
 === "Javascript"
+
     ```js
-    ampli.client.add(yourPlugin())
+    export class MyDestinationPlugin {
+      name = 'my-destination-plugin
```
Comment (Mercy811) [inline docs/data/ampli/plugin.md:36]:
```suggestion
      name = 'my-destination-plugin'
```

File: docs/data/ampli/plugin.md [hunk 3]
```diff
@@ -2,38 +2,209 @@
...
+            code: response.status,
+            event: event,
+            message: response.statusText,
+          };
+       });
+      }
+    }
+    ```
+
+### Hotjar Example
```
Comment (Mercy811) [inline docs/data/ampli/plugin.md:126]: Might be helpful to explain what the Hotjar plugin does, for example

The following code snippet provides a customized Amplitude destination plugin example that can be used to send events to Hotjar by using [hotjar browser SDK](https://www.npmjs.com/package/@hotjar/browser)

File: docs/data/ampli/plugin.md [hunk 4]
```diff
@@ -2,38 +2,209 @@
...
-### Plugin.execute
+## Plugin.execute
 
 This method contains the logic for processing events and has event as parameter. If used as enrichment type plugin, the expected return value is the modified/enriched event. If used as a destination type plugin, the expected return value is a map with keys: `event` (BaseEvent), `code` (number), and `message` (string). This method is called for each event, including Identify, GroupIdentify and Revenue) instrumented using the client interface.
 
 Add plugin to Ampli via `ampli.client.add()`. You can add as many plugin as you like. Each plugin runs in the order based on the plugin type.
 
-=== "Typescript"
-    ```js
+=== "Typescript/Javascript"
```
Comment (justin-fiedler) [inline docs/data/ampli/plugin.md:23]: This state is shared across all code sections. I think it would be better to add another section for JS:

```
=== "TypeScript"
    ```ts
    ampli.client.add(yourPlugin())
    ```

=== "JavaScript"
    ```js
    ampli.client.add(yourPlugin())
    ```
```

Comment (justin-fiedler) [reply docs/data/ampli/plugin.md:23]: Also please check that spelling is consistent or the toggles might not work as expected.

File: docs/data/ampli/plugin.md (hunk 1 above)
Comment (justin-fiedler) [reply docs/data/ampli/plugin.md:8]: +1

**Review by justin-fiedler — CHANGES_REQUESTED** [review]
Thanks @jooohhn. Overall looks good. See nits.

File: docs/data/ampli/plugin.md [hunk 5]
```diff
@@ -11,31 +10,210 @@ template: guide.html
...
+          const hotjarId = user_id || device_id || ""
+          hj.identify(hotjarId, user_properties || {})
+        } else {
+          hj.event(event.event_type)
+        }
+
+        return {
+          code: 0,
+          event: event,
+          message: "hi"
```
Comment (justin-fiedler) [inline docs/data/ampli/plugin.md:166]:
```suggestion
          message: 'Event forwarded to Hotjar API',
```

**Review by justin-fiedler — APPROVED** [review]
